src/observerClass.ts

```diff
@@ -20,11 +21,22 @@ export function makeClassComponentObserver(
     componentClass: React.ComponentClass<any, any>
 ): React.ComponentClass<any, any> {
     const target = componentClass.prototype
-    if (componentClass["isMobXReactObserver"] === true) {
-        console.warn(
-            `The provided component class (${getDisplayName(this)}) is already an observer component.`
-        );
-        return componentClass;
+
+    if (__DEV__) {
+        if (componentClass[mobxObserverProperty] && this.render[mobxAdminProperty]) {
+            console.warn(
+                `The provided component class (${getDisplayName(this)}) is already an observer component.`
+            );
+            return componentClass;
+        } else if (componentClass[mobxObserverProperty]) {
```

Comment:
So this block replaces the one you have deleted down below? Just making sure I understand it correctly.

Reply:
Moves the reactive render check out of the `componentWilllUnmount` block and also performs the `isMobXReactObserve` check, all in one block.  I'll remove the WIP label in my next push, still working things out.